include/alpaka/math/arg/Traits.hpp

```diff
@@ -0,0 +1,60 @@
+/* Copyright 2021 Sergei Bastrakov
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/Common.hpp>
+#include <alpaka/core/Concepts.hpp>
+#include <alpaka/core/Unused.hpp>
+
+#include <complex>
+
+namespace alpaka
+{
+    namespace math
+    {
+        struct ConceptMathArg
+        {
+        };
+
+        //! The math traits.
+        namespace traits
+        {
+            //! The arg trait.
+            template<typename T, typename TArgument, typename TSfinae = void>
+            struct Arg
+            {
+                //!@todo why is this needed here, but not in other functions?
+                ALPAKA_NO_HOST_ACC_WARNING
```

Comment:
I must be missing something very simple. But i have no idea why is there a warning generated here about host and device functions (unless I add a guard). The function works as intended, as tests show and I also checked manually. And in similar situations other functions do not generate such a warning. If anyone has an idea, it's welcome.

Btw the same is for another new math function/trait added in this PR, `conj()`